Cod3rsGrowth.Forms/CadastroEmpresa.cs

```diff
@@ -27,7 +42,16 @@ private void aoClicarDeveAdicionar_Click(object sender, EventArgs e)
                 cnpjEmpresa.TextMaskFormat = MaskFormat.ExcludePromptAndLiterals;
                 empresa.CNPJ = cnpjEmpresa.Text;
                 empresa.Ramo = (EnumRamoDaEmpresa)ramoDaEmpresa.SelectedIndex;
-                _servicoEmpresa.Adicionar(empresa);
+                const int quantoNaoTemId = 0;
+
+                if (_id > quantoNaoTemId)
```

Comment:
faz um método auxiliar, com o nome de algo como "salvarDados", e nele vc faz essa tratativa de se tiver id adiciona, se nao vc atualiza, e retorna, pq vai ficar mais legível e menor, e dá p retornar sem preocupar cm o this.close